pkg/cmd/upgrade/upgrade_boot.go

```
@@ -0,0 +1,442 @@
+		return errors.Wrapf(err, "failed to commit excluded files %v", excludedFiles)
+	}
+	return nil
+}
+
+func (o *UpgradeBootOptions) raisePR() error {
+	gitInfo, err := o.Git().Info(o.Dir)
+	if err != nil {
+		return errors.Wrap(err, "failed to get git info")
+	}
+
+	provider, err := o.gitProvider(gitInfo)
+	if err != nil {
+		return errors.Wrap(err, "failed to get git provider")
+	}
+
+	upstreamInfo, err := provider.GetRepository(gitInfo.Organisation, gitInfo.Name)
+	if err != nil {
+		return errors.Wrapf(err, "getting repository %s/%s", gitInfo.Organisation, gitInfo.Name)
+	}
+
+	details := gits.PullRequestDetails{
+		BranchName: fmt.Sprintf("jx_boot_upgrade"),
+		Title:      "feat(config): upgrade configuration",
+		Message:    "Upgrade configuration",
+	}
+
+	filter := gits.PullRequestFilter{
+		Labels: []string{
+			boot.PullRequestLabel,
+		},
+	}
+	_, err = gits.PushRepoAndCreatePullRequest(o.Dir, upstreamInfo, nil, "master", &details, &filter, false, details.Title, true, false, o.Git(), provider, []string{boot.PullRequestLabel})
+	if err != nil {
+		return errors.Wrapf(err, "failed to create PR for base %s and head branch %s", "master", details.BranchName)
+	}
+	return nil
+}
+
+func (o *UpgradeBootOptions) deleteLocalBranch(branch string) error {
+	err := o.Git().Checkout(o.Dir, "master")
+	if err != nil {
+		return errors.Wrapf(err, "failed to checkout master branch")
+	}
+	err = o.Git().DeleteLocalBranch(o.Dir, branch)
+	if err != nil {
+		return errors.Wrapf(err, "failed to delete local branch %s", branch)
+	}
+	return nil
+}
+
+func (o *UpgradeBootOptions) cloneDevEnv() error {
+	jxClient, devNs, err := o.JXClientAndDevNamespace()
+	devEnv, err := kube.GetDevEnvironment(jxClient, devNs)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get dev environment in namespace %s", devNs)
+	}
+	devEnvURL := devEnv.Spec.Source.URL
+
+	cloneDir, err := ioutil.TempDir("", "")
+	err = os.MkdirAll(cloneDir, util.DefaultWritePermissions)
```

This error is not handled.